Automattic/wp-calypso
# My Home: Update return copy for site title guided tour to reference My Home

#### Changes proposed in this Pull Request

* Updates "Return to the checklist" to say "Return to My Home" for more consistent naming in navigation
* Do we need to include `context` in the translation for the section name `My Home`?

**Before**

<img width="431" alt="Screen Shot 2020-03-25 at 9 59 42 AM" src="https://user-images.githubusercontent.com/2124984/77544446-61dd0c80-6e7f-11ea-86bf-92a2a7f20bd9.png">
<img width="752" alt="Screen Shot 2020-03-25 at 9 59 32 AM" src="https://user-images.githubusercontent.com/2124984/77544450-630e3980-6e7f-11ea-9184-a5b5038a889e.png">

**After**

<img width="750" alt="Screen Shot 2020-03-25 at 10 00 47 AM" src="https://user-images.githubusercontent.com/2124984/77544544-846f2580-6e7f-11ea-9f5e-7f7951a9dd1f.png">
<img width="439" alt="Screen Shot 2020-03-25 at 10 32 35 AM" src="https://user-images.githubusercontent.com/2124984/77547759-fc3f4f00-6e83-11ea-8d32-7c10bfe8fa3a.png">


#### Testing instructions

* Switch to this PR
* Activate the guided tour by going to `/settings/general/{site}/?tour=checklistSiteTitle`
* The first prompt has a return link that should say "Return to My Home"
* Update the site title and follow the remaining prompts
* The last prompt's primary action should read "Return to My Home" rather than "Return to the checklist"
